src/firefly/test/edu/caltech/ipac/firefly/util/FileLoader.java

```diff
@@ -1,40 +1,109 @@
+ * is in the package "edu.caltech.ipac.visualize.plot".  The testing data will be located in the same
+ * package under the test tree.
+ *
  */
-public  class  FileLoader {//extends TestCase {
-    private static final String TEST_DATA_ROOT = "firefly_test_data/";
+public  class  FileLoader {
 
+    private static final String TEST_DATA_ROOT = "firefly_test_data/";
 
+    /**
+     * This method returns the data path for where the test class is located and where is the data is stored.
+     * @param cls
+     * @return
+     * @throws ClassNotFoundException
+     */
     public static String getDataPath(Class cls) throws ClassNotFoundException {
-        //the test class
-       // Class cls = Class.forName(className);
-        //the test subtree path
+
         String testTreePath =  cls.getProtectionDomain().getCodeSource().getLocation().getPath();
         String  rootPath = testTreePath.split("firefly")[0];
         String testDataPath = TEST_DATA_ROOT+cls.getCanonicalName().replaceAll("\\.", "/")
-                .replace(cls.getSimpleName(), "") + File.separatorChar;
-
+                .replace(cls.getSimpleName(), "");
 
         String dataPath = rootPath+ testDataPath;
         return dataPath;
     }
 
-    public static FitsRead loadFitsFile(Class cls, String fitsFile) throws ClassNotFoundException, FitsException {
 
-        String inFitsName =getDataPath(cls)+fitsFile;
+    public static FitsRead loadFitsRead(Class cls, String fitsFile)  {
+
+        try {
+            String inFitsName = getDataPath(cls) + fitsFile;
+
+            Fits fits = new Fits(inFitsName);
+            return FitsRead.createFitsReadArray(fits)[0];
+        }
+        catch (Exception e){
+            e.printStackTrace();
+            return null;
+        }
+
+    }
+
+    public static FitsRead[] loadFitsReadArray(Class cls, String fitsFile)  {
+
+        try {
+            String inFitsName = getDataPath(cls) + fitsFile;
+
+            Fits fits = new Fits(inFitsName);
+            return FitsRead.createFitsReadArray(fits);
+        }
+        catch (Exception e){
+            e.printStackTrace();
+            return null;
+        }
+
+    }
+
+    public static Fits loadFits(Class cls, String fitsFile)  {
+
+        try {
+            String inFitsName = getDataPath(cls) + fitsFile;
+
+            return   new Fits(inFitsName);
+
+        }
+        catch (Exception e){
+            e.printStackTrace();
+            return null;
+        }
+
+    }
+    public static DataGroup loadIpacTable(Class cls, String tblFile) {
+
+        try {
+            File inFile = new File(getDataPath(cls) + tblFile);
+
+            return IpacTableReader.readIpacTable(inFile, null, false, "inputTable");
+        }
+        catch (Exception e){
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+     public static File loadFile(Class cls, String tblFile)  {
```

Comment:
This function is poorly named.  Should be resolveFile(Class cls, String fname) or something like it.
You are not loading anything and it has nothing to do with table.